src/Internal/Marshaller/TypeFactory.php

```diff
@@ -138,5 +139,9 @@ private function getDefaultMatchers(): iterable
         yield DateIntervalType::class;
         yield ArrayType::class;
         yield ObjectType::class;
+
+        if (PHP_VERSION_ID >= 80104) {
+            yield EnumType::class;
+        }
```

Comment:
It preffer to place it before the `ObjectType`.
Please move the block to the top

Reply:
It need to cover by a test also

Reply:
I'll do it.
I've detected some bugs in EnumType and i'll fix it.